File: koopmans/lib/ql/ast/statement/computed_question.rb
module QL
  module AST
    class ComputedQuestion < Question
      attr_reader :assignment

      def initialize(label, variable, type, assignment)
        super(label, variable, type)
        @assignment = assignment
      end

      def accept(visitor, argument = nil)
        visitor.visit_computed_question(self, argument)
      end
    end
  end
end

File: Loading/QL/src/QL/ast/ComputedQuestion.java
package QL.ast;

import QL.ast.expression.Expression;
import QL.ast.type.Type;

public class ComputedQuestion extends Question {

	private final Expression computedQuestion;

	public ComputedQuestion(String variable, String label, Type type, Expression computedQuestion, int line) {
		super(variable, label, type, line);
		this.computedQuestion = computedQuestion;
	}

	public Expression getComputedQuestion() {
		return computedQuestion;
	}

	@Override
	public void accept(FormVisitor v) {
		v.visit(this);
	}
}
